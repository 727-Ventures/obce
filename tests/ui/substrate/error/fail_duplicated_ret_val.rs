use obce::substrate::pallet_contracts::chain_extension::RetVal;

#[obce::error]
enum Error {
    #[obce(ret_val = "100")]
    One,

    #[obce(ret_val = "100")]
    Two
}

fn assert_encode_holds<T: scale::Encode>(_: T) {}
fn assert_try_from_holds<T>(_: T) where RetVal: TryFrom<T> {}

fn main() {
    assert_encode_holds(Error::One);
    assert_try_from_holds(Error::One);
}
