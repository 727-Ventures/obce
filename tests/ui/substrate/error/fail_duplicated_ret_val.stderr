error: ret_val value is used twice
 --> tests/ui/substrate/error/fail_duplicated_ret_val.rs:8:22
  |
8 |     #[obce(ret_val = "100")]
  |                      ^^^^^

error[E0433]: failed to resolve: use of undeclared type `Error`
  --> tests/ui/substrate/error/fail_duplicated_ret_val.rs:16:25
   |
16 |     assert_encode_holds(Error::One);
   |                         ^^^^^ use of undeclared type `Error`

error[E0433]: failed to resolve: use of undeclared type `Error`
  --> tests/ui/substrate/error/fail_duplicated_ret_val.rs:17:27
   |
17 |     assert_try_from_holds(Error::One);
   |                           ^^^^^ use of undeclared type `Error`
